renderdoc/driver/vulkan/wrappers/vk_cmd_funcs.cpp

```diff
@@ -1670,6 +1670,9 @@ VkResult WrappedVulkan::vkEndCommandBuffer(VkCommandBuffer commandBuffer)
   VkResourceRecord *record = GetRecord(commandBuffer);
   RDCASSERT(record);
 
+  if(IsCaptureMode(m_State))
+    AddPendingCommandBufferCallbacks(commandBuffer);
```

Comment:
We shouldn't add these to the global list until the command buffer is submitted to the queue and we know it will eventually complete. Otherwise if the command buffer isn't submitted the event will never be set, and the reference will stick around forever.

Here you need to create & set the event but that should be done internally.